### PR TITLE
feat: add (very) basic back-end monitoring

### PR DESCRIPTION
**Pull request by RomainMuller**
Introduce very basic monitoring for back-end Lambda functions.
Currently this does not trigger any actions, but ensures an
operational dashboard is available with a bird's eye view on
the back-end.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: src/backend/discovery/index.ts [hunk 1]
```diff
@@ -55,5 +62,23 @@ export class Discovery extends Construct {
       schedule: Schedule.rate(timeout),
       targets: [new LambdaFunction(lambda)],
     });
+
+    props.monitoring.watchful.watchLambdaFunction('Discovery Function', lambda);
+    props.monitoring.addHighSeverityAlarm(
```
Comment (eladb) [inline src/backend/discovery/index.ts:67]: High severity means SEV2. I think this is not user-facing

**Review by eladb — CHANGES_REQUESTED** [review]

File: src/backend/discovery/index.ts (hunk 1 above)
Comment (RomainMuller) [reply src/backend/discovery/index.ts:67]: Arguably it is - you have been pushing for us to have a 10 minutes "published in npm --> available on construct hub" SLA, which would make a hiccup there a possible way to violate this SLA.

I don't feel very strongly that this should be a SEV-2 however... the SLA may be a "soft" SLA (aspirational, not contractual)... WDYT?

**Review by iliapolo — APPROVED** [review]

File: src/backend/discovery/index.ts (hunk 1 above)
Comment (eladb) [reply src/backend/discovery/index.ts]: Still don't feel like a SEV-2 and also doesn't feel like the right monitor for this SLA. If we want to manage this SLA we need a dedicated monitor, and even then, I wouldn't say it's SEV-2.

**Review by eladb — COMMENTED** [review]
@RomainMuller where is the part that we actually associate these alarms to actions?

**Pull request merged**